app/post_metadata.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"image"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/dyatlov/go-opengraph/opengraph"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/utils"
+	"github.com/mattermost/mattermost-server/utils/markdown"
+)
+
+const LINK_CACHE_SIZE = 10000
+const LINK_CACHE_DURATION = 3600
+
+var linkCache = utils.NewLru(LINK_CACHE_SIZE)
+
+func (a *App) InitPostMetadata() {
+	// Dump any cached links if the proxy settings have changed so image URLs can be updated
+	a.AddConfigListener(func(before, after *model.Config) {
+		if (before.ServiceSettings.ImageProxyType != after.ServiceSettings.ImageProxyType) ||
+			(before.ServiceSettings.ImageProxyURL != after.ServiceSettings.ImageProxyType) {
+			linkCache.Purge()
+		}
+	})
+}
+
+func (a *App) PreparePostListForClient(originalList *model.PostList) *model.PostList {
+	numPosts := len(originalList.Posts)
+
+	list := &model.PostList{
+		Posts: make(map[string]*model.Post, numPosts),
+		Order: originalList.Order, // Note that this uses the original Order array, so it isn't a deep copy
+	}
+
+	posts := make(chan *model.Post, numPosts)
```

Comment:
I'll look at taking this back out since your right that it seems pretty pointless. I added it in after some improved numbers, but I'm pretty sure that was caused by some other changes done at the same time